camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java

```diff
@@ -116,6 +116,9 @@ public boolean getNext(EtlKey key, BytesWritable payload, BytesWritable pKey) th
         bytes = new byte[origSize];
         buf.get(bytes, buf.position(), origSize);
         pKey.set(bytes, 0, origSize);
+      } else {
+        log.debug("Received message with null message.key().");
```

Comment:
This needs to be a WARN category logging.  And you can also log the 'messageAndOffset' to capture more information.